zerver/lib/streams.py

```diff
@@ -240,6 +240,21 @@ def access_stream_common(user_profile: UserProfile, stream: Stream,
     # an error.
     raise JsonableError(error)
 
+def access_stream_for_admin_actions(user_profile: UserProfile, stream: Stream) -> None:
+    try:
+        sub = Subscription.objects.get(user_profile=user_profile, recipient=stream.recipient)
+    except Subscription.DoesNotExist:
+        sub = None
+
+    error = _("This action requires administrative rights.")
+    if sub is not None:
+        if not (user_profile.is_realm_admin or sub.is_stream_admin):
+            raise JsonableError(error)
+    else:
+        if not user_profile.is_realm_admin:
+            raise JsonableError(error)
+    return
```

Comment:
I think this would read more clearly if it had `if user_profile.is_realm_admin: return` earlier rather than doing that check multiple times.